cpp/src/arrow/extension/fixed_shape_tensor.cc

```diff
@@ -18,7 +18,9 @@
 #include <numeric>
 #include <sstream>
 
+#include <arrow/scalar.h>
```

Comment:
Can we avoid the gratuitous style inconsistency?
```suggestion
#include "arrow/scalar.h"
```

(will need to reformat afterthis)